lib/pdk/module/build.rb

```diff
@@ -196,6 +196,13 @@ def ignored_files
 
                              PathSpec.new(data)
                            end
+
+        # Also ignore the target directory if it is in the module dir and not already ignored
+        if Find.find(@module_dir).include?(@target_dir) && !@ignored_files.match(File.basename(@target_dir) + '/')
```

Comment:
Rather than calling `Find.find` here (which might get expensive), could we get away with something like:

```ruby
if @ignored_files.specs_matching(target_dir).empty?
  @ignored_files.add("\/#{File.basename(target_dir)\/")
end
```

Reply:
That wouldn't guard against the scenario where `target-dir` wasn't a sub directory of the `module_dir`. If `target-dir` was `/tmp/pkg/` then `@ignored_files.specs_matching('/tmp/pkg/').empty?` would be `true`, and `/pkg/` would get added to the ignored_files a second time.